datadog_checks_base/datadog_checks/base/utils/http.py

```diff
@@ -630,6 +635,73 @@ def read(self, **request):
             return self._token
 
 
+class DCOSAuthTokenReader(object):
+    def __init__(self, config):
+        self._login_url = config.get('login_url', '')
+        if not isinstance(self._login_url, str):
+            raise ConfigurationError('The `login_url` setting of DC/OS auth token reader must be a string')
+        elif not self._login_url:
+            raise ConfigurationError('The `login_url` setting of DC/OS auth token reader is required')
+
+        self._service_account = config.get('service_account', '')
+        if not isinstance(self._service_account, str):
+            raise ConfigurationError('The `service_account` setting of DC/OS auth token reader must be a string')
+        elif not self._service_account:
+            raise ConfigurationError('The `service_account` setting of DC/OS auth token reader is required')
+
+        self._private_key_path = config.get('private_key_path', '')
+        if not isinstance(self._private_key_path, str):
+            raise ConfigurationError('The `private_key_path` setting of DC/OS auth token reader must be a string')
+        elif not self._private_key_path:
+            raise ConfigurationError('The `private_key_path` setting of DC/OS auth token reader is required')
+
+        self._expiration = config.get('expiration', 300)  # default to 5 minutes
+        if not isinstance(self._expiration, int):
+            raise ConfigurationError('The `expiration` setting of DC/OS auth token reader must be an integer')
+
+        self._token = None
+
+    def read(self, **request):
+        if self._token is None or 'error' in request:
+            with open(self._private_key_path, 'rb') as f:
+                global default_backend
```

Comment:
I don't get these globals and conditional imports. Why not just import them right at the top of the module?

Reply:
the idea is to do lazy imports per suggestion from @ofek